src/engine/effects/engineeffectchain.cpp

```diff
@@ -308,12 +311,17 @@ bool EngineEffectChain::process(const ChannelHandle& inputHandle,
                     }
 
                     processingOccured = true;
+                    effectChainGroupDelay += pEffect->getGroupDelay();
+
                     // Output of this effect becomes the input of the next effect
                     pIntermediateInput = pIntermediateOutput;
                 }
             }
         }
 
+        m_effectsDelay->setDelay(effectChainGroupDelay);
```

Comment:
We need to be careful not to swap stereo channels.
I think if we make the group delay by frames it is solved automatically.

Reply:
That's a good point. If the effectChainGroupDelay isn't right to the number of channels, the channels could be swapped (for stereo signal an odd number of samples). This situation isn't handled internally for now. As I mentioned for the PitchShiftEffect setGroupDelay() method, it is necessary to have the information about the number of channels, if I wouldn't assume, that it will be worked only with the stereo audio signal.

Reply:
Okay, I agree. If it is ok, the setter will recalculate the number of the frames into samples by using [mixxx::kEngineChannelCount](https://github.com/mixxxdj/mixxx/blob/1cd826af252c51f96db17f8a27cd3599828aa946/src/engine/engine.h#L8), I'm fine with that. The reason why the number of frames has to be recalculated in the setter is, that the inner process() method structure works with samples and it doesn't make sense to recalculate samples for every delay value usage.